tests/http_test_server.py

```diff
@@ -18,6 +18,10 @@ def run_test_server(directory: str) -> http.server.ThreadingHTTPServer:
     """
 
     class DualStackServer(http.server.ThreadingHTTPServer):
+        daemon_threads = False  # These are per-request threads
+        allow_reuse_address = True  # Good for tests
+        request_queue_size = 64  # Should be more than the number of test packages
```

Comment:
I'm guessing request_queue_size < number of conda request threads is the real problem. It doesn't hurt to have a long queue.

allow_reuse_address is unchanged from the superclass default.

If daemon_threads keeps the test suite alive for noticeably longer (seems unlikely but who knows) we can take it out.